collector/zfs_linux.go

```diff
@@ -81,6 +98,50 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 	return scanner.Err()
 }
 
-func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+func (c *zfsCollector) parsePoolProcfsFiles(zpoolPaths []string, handler func(string, zfsSysctl, int)) (err error) {
+
+	for _, zpoolPath := range zpoolPaths {
+		file, err := os.Open(zpoolPath)
+		if err != nil {
+			log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", zpoolPath)
+			err = zfsNotAvailableError
+		}
+		defer file.Close()
```

Comment:
After reviewing the rest of this, yeah, I'd definitely recommend breaking out a scanning function to reduce indentation and improve readability.